src/store/ShareStore.js

```diff
@@ -120,7 +123,7 @@ export default class ShareStore {
     @action.bound downloadPNG() {
         this.menu.setOpen(false);
         const root = findAncestor(this.stx.container, 'ciq-chart-area');
-        html2canvas(root).then((canvas) => {
+        html2canvas(root).then((canvas) => { // eslint-disable-line no-undef
```

Comment:
it could look like this,

```javascript
loadHtml2Canvas() {
     if(this._promise_html2canas) {
             return this._promise_html2canvas;
      }
      this._promise_html2canvas = new PendingPromise();
      CIQ.loadScript(html2canvasCDN, () => this._promise_html2canvas.resolve());
}
// later use it in 
@action.bound downloadPNG() {
     // ....
     loadHtml2Canvas()
       .then(() => window.html2canvas(root))
       .then((canvas) => { // ...
       })
}
```

Reply:
do we really need a promise for that? Looks more like an extra step

Reply:
@brucebinary well, without a promise if you open and close the menu quickly ... you can end up loading the library twice.

have a look at `loadScript` implementation.

Reply:
ok just added that in.